src/configParser.js

```diff
@@ -76,6 +79,26 @@ ConfigParser.prototype._mergeConfig = function (configFile, type) {
   this.config = _mergeWithArrays(newConfig, priorityConfig);
 };
 
+ConfigParser.prototype._resolveProfile = function (profile, fullConfig) {
+  fullConfig = fullConfig || [];
+  if (!profile) {
+    return [];
+  }
+  var fileConfig;
+  try {
+    fileConfig = _readConfig(CONFIG_DIR + profile + '.profile.conf.js');
+  } catch (e) {
```

Comment:
handling exceptions as part of normal processing is an antipattern IMHO.  I would personally return null for file-not-found and handle it respectively. Of course, any error will throw an exception.

Reply:
replaced this with a sync check returning a boolean.
unfortunately now the tests fail on ci, altough they are ok locally.
this is again not a good idea: it's strongly discouraged to check if a file exists, and then read from it, as the file may be deleted in the meantime. but all other options involve a try/catch